android/app/src/main/java/com/sample/react/ReactBundleNameProvider.kt

```diff
@@ -0,0 +1,24 @@
+package com.sample.react
+
+import android.content.Context
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private val possibleEntryFiles = listOf(
```

Comment:
I'd move this into the function. It's not used elsewhere.

Reply:
The reason to move it out is to avoid allocating the list every time the function is called (internally this list will be "java static"). The variable is also private to the file, so I would rather keep it as it is.

Reply:
Correct me if I'm wrong but doesn `lazy` imply that it only gets calculated once? I'd rather not have a list in static memory if possible.

Reply:
That's correct, `lazy` in combination with the `ReactBundleNameProvider` being a singleton guarantee that the method will be executed once (unless you explicitly re-init the dependency graph).

The list itself, as well as its contents are immutable. It is safe to have it as static afaik. I will move it down to the method definition, but the difference here mostly comes down to the personal preference :P